app_book/views.py

```diff
@@ -311,3 +312,44 @@ def decline_counteroffer(request, id):
                          'Der Preisvorschlag wurde erfolgreich abgelehnt. Der Interessent wird benachrichtigt')
 
     return HttpResponseRedirect(reverse('app_notification:notificationsPage'))
+
+def books(request):
+    template_name = 'app_book/books.html'
+
+    order_dir = request.GET.get('order_dir', '')
+    order_by = request.GET.get('order_by', '')
+    page = request.GET.get('page')
+
+    if order_dir == 'desc':
+        direction = '-'
+    elif order_dir == 'asc':
+        direction = ''
+    else:
+        direction = ''
+
+    if order_by == 'date':
```

Comment:
Ich würde den Namen in `order_by` genauso wie das Attribut nennen. Dadurch ersparst du dir dieses Mapping. Musst aber natürlich prüfen, dass nur erlaubte Attribute angegeben werden.

```
offer_list = Offer.objects.filter(active=True).order_by(direction + 'book__' + order_by)
```